janus_server/src/datastore.rs

```diff
@@ -127,63 +129,180 @@ impl Transaction<'_> {
+            )
+            .await?;
+        self.tx
+            .execute(
+                &stmt,
+                &[
+                    /* task_id */ &&task.id.0[..],
+                    /* config_id */ &config_ids,
+                    /* configs */ &configs,
+                    /* private_keys */ &private_keys,
+                ],
+            )
+            .await?;
+
         Ok(())
     }
 
+    /// Fetch the task parameters corresponing to the provided `task_id`.
+    //
+    // Only available in test configs for now, but will soon be used by
+    // aggregators to discover tasks from the database.
+    #[cfg(test)]
+    pub(crate) async fn get_task_by_id(&self, task_id: TaskId) -> Result<TaskParameters, Error> {
+        use postgres_types::ToSql;
+
+        let params: &[&(dyn ToSql + Sync)] = &[&&task_id.0[..]];
+        let stmt = self
+            .tx
+            .prepare_cached(
+                "SELECT aggregator_role, aggregator_endpoints, vdaf, vdaf_verify_param,
+                max_batch_lifetime, min_batch_size, min_batch_duration, tolerable_clock_skew,
+                collector_hpke_config, agg_auth_key
+                FROM tasks WHERE task_id = $1",
+            )
+            .await?;
+        let task_row = self.tx.query(&stmt, params);
+
+        let stmt = self
+            .tx
+            .prepare_cached(
+                "SELECT config_id, config, private_key FROM task_hpke_keys
+                WHERE task_id = (SELECT id FROM tasks WHERE task_id = $1)",
+            )
+            .await?;
+        let hpke_key_rows = self.tx.query(&stmt, params);
+
+        let (task_row, hpke_key_rows) = try_join(task_row, hpke_key_rows).await?;
+        let task_row = single_row(task_row)?;
+
+        self.task_parameters_from_rows(task_id, task_row, hpke_key_rows)
+    }
+
+    /// Fetch all the tasks in the database.
+    #[cfg(test)]
+    pub(crate) async fn get_tasks(&self) -> Result<Vec<TaskParameters>, Error> {
+        use std::collections::HashMap;
+
+        let stmt = self
+            .tx
+            .prepare_cached(
+                "SELECT task_id, aggregator_role, aggregator_endpoints, vdaf,
+                vdaf_verify_param, max_batch_lifetime, min_batch_size, min_batch_duration,
+                tolerable_clock_skew, collector_hpke_config, agg_auth_key 
+                FROM tasks",
+            )
+            .await?;
+        let task_rows = self.tx.query(&stmt, &[]);
+
+        let stmt = self
+            .tx
+            .prepare_cached(
+                "SELECT (SELECT tasks.task_id FROM tasks WHERE tasks.id = task_hpke_keys.task_id),
+            config_id, config, private_key FROM task_hpke_keys",
+            )
+            .await?;
+        let hpke_config_rows = self.tx.query(&stmt, &[]);
+
+        let (task_rows, hpke_config_rows) = try_join(task_rows, hpke_config_rows).await?;
+
+        let mut task_row_by_id = HashMap::new();
```

Comment:
Nit, take it or leave it: I think this could be a `Vec<(TaskId, Row)>`, as we only iterate over it later.

Reply:
You're correct, and I suppose a `Vec` is slightly more efficient.